lib/options.go

```diff
@@ -242,6 +341,9 @@ type Options struct {
 	// Blacklist IP ranges that tests may not contact. Mainly useful in hosted setups.
 	BlacklistIPs []*IPNet `json:"blacklistIPs" envconfig:"K6_BLACKLIST_IPS"`
 
+	// Block hostname patterns that tests may not contact.
+	BlockedHostnames *HostnameTrie `json:"blockHostnames" envconfig:"K6_BLOCK_HOSTNAMES"`
```

Comment:
Ah, this is somewhat problematic. Because of a bug in the `envconfig` library, it initializes `BlockedHostnames`, so the `if opts.BlockedHostnames != nil` check in `Apply()` will always be true. So, you're never able to set the blocked hostnames in the JS script, because `Apply` will always overwrite them, even if it's with an empty list. 

Try running this script with k6:
```js
import http from "k6/http";

export let options = {
    throw: true,
    blockHostnames: ["*.io"],
};

export default function () {
    http.get("https://test.k6.io/");
}
```

k6 will not block the domain and the request will happen! :disappointed: The proper fix for this would be to replace the envconfig library with something saner, but that's a huge refactoring task we've been postponing for a long time... :sob: For now, maybe we shouldn't use a pointer here, and instead have the pointer `HostnameTrie` struct? Or, maybe remove `envconfig:"K6_BLOCK_HOSTNAMES"` and add support for environment variables whenever we fix envconfig?

See https://github.com/kelseyhightower/envconfig/issues/113 and https://github.com/loadimpact/k6/issues/1560 and other [envconfig issues](https://github.com/loadimpact/k6/issues?q=is%3Aissue+is%3Aopen+envconfig).

Reply:
envconfig strikes again. I will experiment with best options here since having configuration through environment variables is desirable for this sort of thing. Worst comes to worst, I will remove the envconfig support.